plutarch-trace/src/Plutarch/TraceSig.hsig

```diff
@@ -0,0 +1,18 @@
+signature Plutarch.TraceSig (ptrace', ptrace, ptraceIfTrue, ptraceIfFalse) where
+
+import Prelude ()
+import Plutarch.Prelude (Term, (:-->))
+import Plutarch.String (PString)
+import Plutarch.Bool (PBool)
+
+-- | A strict version of 'ptrace'.
+ptrace' :: Term s (PString :--> a :--> a)
+
+-- | Trace a message, then evaluate and return given argument.
+ptrace :: Term s PString -> Term s a -> Term s a
+
+-- | Trace a message if given argument evaluates to true.
+ptraceIfTrue :: Term s (PString :--> PBool :--> PBool)
```

Comment:
Not sure whether the signatures were originally like this, but we probably want to make this `Term s PString -> Term s PBool -> Term s PBool`, otherwise we can't elide the string after all.

Reply:
Do you think that may be a trade-off? (wondering about preferring Plutarch level functions - though I do get the advantage of elision)

Reply:
It is a trade-off, but in this case it's best if we do it this way.

Reply:
Do note that you can still hoist it even if it's a Haskell level function by making a helper Plutarch-level function which you hoist.

Reply:
Done!

Reply:
Ok, done for real. I do wonder though, should the non tracing versions still evaluate their argument (not the string one)?

Reply:
They should just pass it through, which is the same as applying it to `id`.